File: Trees/BinaryTreeNode.py
__author__='Kalpak Seal'

class BinaryTreeNode:
    def __init__(self, data):
        self.data = data
        self.left = None
        self.right = None

    def set_data(self, data):
        self.data = data

    def get_data(self):
        return self.data

    def get_left(self):
        return self.left

    def get_right(self):
        return self.right


if __name__ == '__main__':
    root = BinaryTreeNode(1)

    root.left = BinaryTreeNode(2)
    root.right = BinaryTreeNode(3)

    root.left.left = BinaryTreeNode(4)
    root.left.right = BinaryTreeNode(5)

    root.right.left = BinaryTreeNode(6)
    root.right.right = BinaryTreeNode(7)

    print (root.get_data())
    print (root.left.get_data())
    print (root.right.get_data())
    print (root.left.left.get_data())
    print (root.left.right.get_data())
    print (root.right.left.get_data())
    print (root.right.right.get_data())

File: Trees/1. Binary Tree implementation and traversals/BinaryTreeNode.java
public class BinaryTreeNode
{
    private int data;
    private BinaryTreeNode left, right;

    public BinaryTreeNode(int data)
    {
        this.data = data;
        this.left = null;
        this.right = null;
    }

    public int getData()
    {
        return data;
    }

    public void setData(int data)
    {
        this.data = data;
    }

    public BinaryTreeNode getLeft()
    {
        return left;
    }

    public void setLeft(BinaryTreeNode left)
    {
        this.left = left;
    }

    public BinaryTreeNode getRight()
    {
        return right;
    }

    public void setRight(BinaryTreeNode right)
    {
        this.right = right;
    }
}